src/compas/datastructures/mesh/core/halfedge.py

```diff
@@ -49,100 +42,44 @@ class HalfEdge(Datastructure):
     @property
     def DATASCHEMA(self):
         import schema
-        if LooseVersion(compas.__version__) < LooseVersion('0.16.5'):
-            return schema.Schema({
-                "attributes": dict,
-                "dva": dict,
-                "dea": dict,
-                "dfa": dict,
-                "vertex": schema.And(
-                    dict,
-                    lambda x: all(isinstance(key, int) for key in x)
-                ),
-                "face": schema.And(
-                    dict,
-                    lambda x: all(isinstance(key, int) for key in x),
-                    lambda x: all(all(isinstance(item, int) for item in value) for value in x.values())
-                ),
-                "facedata": schema.And(
-                    dict,
-                    lambda x: all(isinstance(key, int) for key in x),
-                    lambda x: all(isinstance(value, dict) for value in x.values())
-                ),
-                "edgedata": dict,
-                "max_int_key": schema.And(int, lambda x: x >= -1),
-                "max_int_fkey": schema.And(int, lambda x: x >= -1)
-            })
         return schema.Schema({
-            "compas": str,
-            "datatype": str,
-            "data": {
-                "attributes": dict,
-                "dva": dict,
-                "dea": dict,
-                "dfa": dict,
-                "vertex": schema.And(
-                    dict,
-                    lambda x: all(isinstance(key, int) for key in x)
-                ),
-                "face": schema.And(
-                    dict,
-                    lambda x: all(isinstance(key, int) for key in x),
-                    lambda x: all(all(isinstance(item, int) for item in value) for value in x.values())
-                ),
-                "facedata": schema.And(
-                    dict,
-                    lambda x: all(isinstance(key, int) for key in x),
-                    lambda x: all(isinstance(value, dict) for value in x.values())
-                ),
-                "edgedata": schema.And(
-                    dict,
-                    lambda x: all(isinstance(key, str) for key in x),
-                    lambda x: all(isinstance(value, dict) for value in x.values())
-                ),
-                "max_vertex": schema.And(int, lambda x: x >= -1),
-                "max_face": schema.And(int, lambda x: x >= -1)
-            }
+            "attributes": dict,
+            "dva": dict,
+            "dea": dict,
+            "dfa": dict,
+            "vertex": schema.And(
+                dict,
+                lambda x: all(isinstance(key, int) for key in x),
+                lambda x: all(key >= 0 for key in x),
```

Comment:
This looks like it gets reused often.  Maybe adding a `is_sequence_of_indices` method would be easier to read.